tools/c7n_azure/c7n_azure/resources/arm_resource.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_azure.provider import resources
+from c7n_azure.resources.arm import ArmResourceManager, arm_resource_types
+from c7n_azure.utils import ResourceIdParser
+from c7n.filters.core import Filter, type_schema
+
+
+@resources.register('armresource')
+class ArmResource(ArmResourceManager):
```

Comment:
We could just go with GenericResource now that you refactored this?  The Base makes sense as ARM Resource Base but this one is the actual GenericResource.

I'm not sure if this convention actually exists in the API, but it is common in the tooling.